library/general/src/lib/y2user/readers/users_simple.rb

```diff
@@ -0,0 +1,26 @@
+require "yast2/execute"
+require "date"
+
+require "y2user/group"
+require "y2user/user"
+require "y2user/password"
+
+Yast.import "UsersSimple"
+
+module Y2User
+  module Readers
+    # Reads users configuration using old Yast Module UsersSimple.
+    class UsersSimple
+      def read_to(configuration)
+        users = Yast::UsersSimple.GetUsers
+        # TODO: only created users, not imported ones for now
+        users.each do |user|
+          configuration.users << User.new(configuration, user["uid"], gecos: [user["cn"]])
```

Comment:
when writing this code I get idea that because configuration is passed, then user can auto register itself there. So it do configuration.users << self in constructor. What do you think about this idea?

Reply:
I think it's acceptable

Reply:
Just a question: should we check that the user is not already in the given configuration?

Reply:
@dgdavid well, my idea is to have some validation method to check it as it is more tricky that it looks like as e.g. uid should not collide, but it is one tricky usage where it can be with e.g. user toor that has uid 0 and can act as root. So question is if user name is unique enough? What if there is ldap and local user with same name? What happen if it happen?

Reply:
> So question is if user name is unique enough? What if there is ldap and local user with same name? 

So, perhaps we should use uid and username as _user primary key_. But not sure.

Reply:
For me it is reasonable. And yes, we need some validations at some point.

BTW, when I was looking at this line, I have started asking myself if we could avoid the double explicit reference between User and Configuration (and the same for Password and Group). I mean, adding an user to a configuration should be enough to know the configuration that the user belongs to. For example, `User#configuration` could be a lazy method that finds the user object in any of the existing configurations. What do you think?

Reply:
Ok, I wrote my comment after reading the code snipped but without reading your comment. Ok, doing it in the User constructor would also avoid to explicitly indicate the double reference.

Reply:
BTW, what happens if we do:

~~~
user = User.new(config1, ...)
config2.users << user
~~~

Is that possible?

Reply:
no, that should be prevented. Correct method is 
```
user = User.new(config1, ...)
user.clone_to(config2)
```

About idea with lazy evaluation I do not like it much as that references should be fast and can happen really often.
But I think some way to prevent such wrong usage would be nice. But maybe that validation is enough?